### PR TITLE
Get more consistent distributions in parallel scenarios test

### PR DESCRIPTION
**Pull request by abadams**
Note that this is not denoising the individual runtimes. The point of the test is to measure a distribution of runtimes for a variety of scenarios. This PR takes more samples for each distribution, and spreads the samples out over time to dodge transitory OS stuff by interleaving different test scenarios.

I also tweaked some of the scenarios to be more useful, and reported actual deciles so that the median is actually there.

### CODE REVIEW

**Review by steven-johnson — APPROVED** [review]

File: test/performance/parallel_scenarios.cpp [hunk 1]
```diff
@@ -31,31 +31,36 @@ int main(int argc, char **argv) {
...
     int native_threads = Halide::Internal::JITSharedRuntime::get_num_threads();
 
+    std::map<std::tuple<bool, bool, int, int>, std::vector<float>> results;
+
     auto bench = [&](bool m, bool c, int i, int o) {
-        const int num_samples = 128;
         const int memory_limit = m ? max_memory : 128;
 
+        auto now = std::chrono::high_resolution_clock::now;
+        auto to_ns = [](auto delta) { return 1e9 * std::chrono::duration<float>(delta).count(); };
```
Comment (mcourteaux) [inline test/performance/parallel_scenarios.cpp:40]: Nitpick: 1e9 is a double, so the lambda returns a double. Either `1e9f`, or the results vector should become double.

**Comment by steven-johnson** [comment]
ready to land?

**Comment by abadams** [comment]
Looks like there was a merge conflict to fix. Apart from that, assuming everything goes green, yes

**Comment by abadams** [comment]
Failures are all vulkan stuff

**Pull request merged**